### PR TITLE
chore(test): add integration tests

### PR DESCRIPTION
**Pull request by sdodson99**
# Description

Setup and add tests so that we can continue to make changes with confidence.

# Changes

* Replaced Jest with Vitest
* Added integration test utilities to render test subject within app providers
* Setup Firebase mocks
* Accidentally increased test coverage to 100% with just 8 tests 😃 

# Q&A

**Why Vitest?**

* Better performance than Jest
* Way easier setup since it leverages Vite's config, transformers, plugins, built-in TS support, etc. 
* Small learning curve since Vitest and Jest have compatible APIs

**Why Integration Testing?**

* Allow us to test our package from the perspective of a client application
* Allow us to write tests with minimal mocks

### CODE REVIEW

File: package/.babelrc
Comment (sdodson99) [inline package/.babelrc:1]: No longer have to go through Babel compilation since Vitest transforms our tests through Vite 🔥

File: package/.parcelrc
Comment (sdodson99) [inline package/.parcelrc:1]: Only reason for this was to override Babel configuration, which were no longer using 🔥

File: package/src/use-comments.test.tsx [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+    renderApp(<TestComponent {...props} />, { contentId });
+
+    const comment1 = await screen.findByText('Comment 1');
+    const comment2 = await screen.findByText('Comment 2');
+    expect(comment1).toBeInTheDocument();
+    expect(comment2).toBeInTheDocument();
+  });
+
+  it('uses comments from client-side comment state', async () => {
```
Comment (edodso2) [inline package/src/use-comments.test.tsx:63]: The tests themselves are user oriented and feel like integration tests but the descriptions seem low level to me. Is this intentional? The alternative would be something like:

`it('displays comments on app load', ...`
`it('displays newly added comment',...`

**Review by edodso2 — APPROVED** [review]
Vitest is awesome! The abstractions in `test/integration/setup` make the tests so clean. I am not sure how the abstraction will fair in the long run but for now its 🔥.

File: package/src/use-comments.test.tsx (hunk 1 above)
Comment (sdodson99) [reply package/src/use-comments.test.tsx]: Good catch. This test in-particular is way too implementation specific. I went through and adjusted most of the test descriptions. Let me know if they could still use improvements tho.

Comment (edodso2) [reply package/src/use-comments.test.tsx]: Looks good to me now 👍

**Pull request merged**